pennylane/qnn/torch.py

```diff
@@ -0,0 +1,311 @@
+            raise ValueError("Must specify a shape for every non-input parameter in the QNode")
+
+        if qnode.func.var_pos:
+            raise TypeError("Cannot have a variable number of positional arguments")
+
+        if qnode.func.var_keyword:
+            raise TypeError("Cannot have a variable number of keyword arguments")
+
+        self.qnode = to_torch(qnode)
+        weight_shapes = {
+            weight: (tuple(size) if isinstance(size, Iterable) else (size,) if size > 1 else ())
+            for weight, size in weight_shapes.items()
+        }
+
+        # Allows output_dim to be specified as an int, e.g., 5, or as a length-1 tuple, e.g., (5,)
+        self.output_dim = output_dim[0] if isinstance(output_dim, Iterable) else output_dim
+
+        defaults = {
+            name for name, sig in self.sig.items() if sig.par.default != inspect.Parameter.empty
+        }
+        self.input_is_default = self.input_arg in defaults
+        if defaults - {self.input_arg} != set():
+            raise TypeError(
+                "Only the argument {} is permitted to have a default".format(self.input_arg)
+            )
+
+        if not init_method:
+            init_method = functools.partial(torch.nn.init.uniform_, b=2 * math.pi)
+
+        self.qnode_weights = {}
+        for name, size in weight_shapes.items():
+            if len(size) == 0:
+                self.qnode_weights[name] = torch.nn.Parameter(init_method(torch.Tensor(1))[0])
+            else:
+                self.qnode_weights[name] = torch.nn.Parameter(init_method(torch.Tensor(*size)))
+            self.register_parameter(name, self.qnode_weights[name])
+
+    def forward(self, inputs):  # pylint: disable=arguments-differ
+        """Evaluates a forward pass through the QNode based upon input data and the initialized
+        weights.
+
+        Args:
+            inputs (tensor): data to be processed
+
+        Returns:
+            tensor: output data
+        """
+        if len(inputs.shape) == 1:
+            return self._evaluate_qnode(inputs)
+
+        return torch.stack([self._evaluate_qnode(x) for x in inputs])
+
+    def _evaluate_qnode(self, x):
+        """Evaluates the QNode for a single input datapoint.
+
+        Args:
+            x (tensor): the datapoint
+
+        Returns:
+            tensor: output datapoint
+        """
+        qnode = self.qnode
+
+        for arg in self.sig:
+            if arg is not self.input_arg:  # Non-input arguments must always be positional
+                w = self.qnode_weights[arg]
+
+                qnode = functools.partial(qnode, w)
+            else:
+                if self.input_is_default:  # The input argument can be positional or keyword
+                    qnode = functools.partial(qnode, **{self.input_arg: x})
+                else:
+                    qnode = functools.partial(qnode, x)
+        return qnode().type(x.dtype)
+
+    def __str__(self):
```

Comment:
Note that the `torch.nn` Module has an `extra_repr` method for adding more details like this, but it kept being shown in the methods section of the generated docs. So I chose this simpler option that we used for the KerasLayer.